File: examples/spring/src/main/java/org/fluentlenium/example/spring/page/MainPage.java
package org.fluentlenium.example.spring.page;

import org.fluentlenium.core.FluentPage;
import org.fluentlenium.core.annotation.PageUrl;
import org.fluentlenium.core.domain.FluentWebElement;
import org.openqa.selenium.support.FindBy;

import static org.fluentlenium.assertj.FluentLeniumAssertions.assertThat;

@PageUrl("/")
public class MainPage extends FluentPage {

    @FindBy(className = "whats-fluentlenium")
    private FluentWebElement content;

   public void verifyIfIsLoaded() {
        assertThat(content).isDisplayed();
   }
}

File: examples/quickstart-firefox/src/main/java/io/fluentlenium/examples/pages/fluentlenium/MainPage.kt
package io.fluentlenium.examples.pages.fluentlenium

import io.fluentlenium.assertj.FluentLeniumAssertions.assertThat
import io.fluentlenium.core.FluentPage
import io.fluentlenium.core.annotation.PageUrl
import io.fluentlenium.core.domain.FluentWebElement
import io.fluentlenium.examples.components.fluentlenium.Header
import org.openqa.selenium.support.FindBy

@PageUrl("https://fluentlenium.io")
class MainPage : FluentPage() {

    fun perform(fn: MainPage.() -> Unit) = this.apply(fn)

    @FindBy(className = "whats-fluentlenium")
    lateinit var mainContent: FluentWebElement

    @FindBy(css = "nav")
    lateinit var header: Header

    override fun isAt() {
        assertThat(mainContent).isDisplayed
    }

}